app/models/experimentation/notifications_experiment.rb

```diff
@@ -42,34 +42,57 @@ def self.eligible_patients
                                              "end_time > ? OR treatment_group_memberships.created_at > ?",
                                              Time.current, MONITORING_BUFFER.ago
                                            ).select(:patient_id))
-        .then { |patients| exclude_bangladesh_blocks(patients) }
-        .then { |patients| include_only_allowed_india_states(patients) }
+        .then { |patients| filter_patients(patients, filters) }
     end
 
-    def self.exclude_bangladesh_blocks(patients)
-      return patients unless CountryConfig.current_country?("Bangladesh")
-
+    def self.filter_patients(patients, filters)
       patients
         .merge(Facility.with_block_region_id)
         .select("patients.*")
-        .where.not(block_region: {id: excluded_block_ids.presence})
+        .then { |patients| filter_states(patients, filters["states"]) }
+        .then { |patients| filter_blocks(patients, filters["blocks"]) }
+        .then { |patients| filter_facilities(patients, filters["facilities"]) }
     end
 
-    def self.include_only_allowed_india_states(patients)
-      return patients unless CountryConfig.current_country?("India") && SimpleServer.env.production?
-      return patients unless ENV["EXPERIMENT_ALLOWED_STATES"].present?
+    def self.filter_states(patients, states)
+      return patients unless states
 
-      states = ENV.fetch("EXPERIMENT_ALLOWED_STATES", "").split(",").map(&:strip)
-      patients.where(facilities: {state: states})
+      if states["include"]
+        patients.where(facilities: {state: states["include"]})
```

Comment:
I'm sticking to names of state instead of UUID as mentioned in regions/reporting_patient_states table because extracting state_id becomes too complicated with the existing`with_block_region_id` scope.

Reply:
State name is used in current design, so I am reusing that logic in new design.